spectacles/runner.py

```diff
@@ -249,6 +264,24 @@ async def get_project_imports(self) -> List[str]:
         else:
             return [p["name"] for p in manifest["imports"] if not p["is_remote"]]
 
+    async def checkout_personal_branch(self, ref: str) -> str:
+        """Updates the user's personal branch to the git ref."""
+        await self.update_workspace("dev")
+        if not self.personal_branch:
+            self.personal_branch = await self.get_personal_branch()
+        await self.client.checkout_branch(self.project, self.personal_branch)
+        await self.client.reset_to_remote(self.project)
+        await self.client.hard_reset_branch(self.project, self.personal_branch, ref)
+        return self.personal_branch
+
+    async def get_personal_branch(self) -> str:
+        """Finds the name of the user's personal branch."""
+        branches = await self.client.get_all_branches(self.project)
+        for branch in branches:
+            if branch["personal"] and not branch["readonly"]:
+                return str(branch["name"])
+        raise ValueError("Personal branch not found")
```

Comment:
Is there any other info we should raise here that might help with debugging? Maybe project?